mono/metadata/marshal-ilgen.c

```diff
@@ -1790,7 +1790,7 @@ emit_native_wrapper_ilgen (MonoImage *image, MonoMethodBuilder *mb, MonoMethodSi
 #endif
 
 		mono_mb_emit_ldloc_addr (mb, coop_gc_stack_dummy);
-		mono_mb_emit_icall (mb, mono_threads_enter_gc_safe_region_unbalanced);
+		mono_mb_emit_icall (mb, ves_icall_mono_threads_enter_gc_safe_region_unbalanced);
 		mono_mb_emit_stloc (mb, coop_gc_var);
```

Comment:
We use ves_icall for real icalls called from managed code, there is no need to rename these imho.

Reply:
Lambdageek like it, as I understand, both from JIT and C#, though I realize JIT usually does not. I wonder if it should be three naming conventions though. I was going to call it foo_jit or foo_from_jit before lambdageek reiterated ves_icall.

I do need a second name now, but foo_internal would suffice (and foo_helper or foo_impl for foo and foo_internal to bottleneck on)

Reply:
Maybe I confused the functions and don't require a new name. Let me see again.

Reply:
Should be able to rename just two functions instead of six..

Reply:
No, there is reason to rename/extend all six. Though for four of them unclear it is worth it. There are calls from JIT and calls from the macros. The macros can be more easily maintained to fill in the function name. The JIT is less statically checked, so I felt it more maintainable to leave more of the work in C instead of generated IL. However the dynamic binding renames with some level of coupling either way. Originally I didn't know how to maintain the ilgen but that is flimsy and moot -- it is mono_emit_ldstr or such.